File: backend/src/main/java/com/taskio/service/BoardService.java
package com.taskio.service;

import com.taskio.dto.request.CreateBoardRequest;
import com.taskio.dto.request.UpdateBoardRequest;
import com.taskio.dto.request.AddMemberRequest;
import com.taskio.dto.response.BoardResponse;
import java.util.List;

public interface BoardService {
    List<BoardResponse> getBoardsByUser(Long userId);

    BoardResponse createBoard(Long userId, CreateBoardRequest request);

    BoardResponse getBoard(Long boardId);

    BoardResponse updateBoard(Long boardId, Long userId, UpdateBoardRequest request);

    void addMember(Long boardId, Long userId, AddMemberRequest request);
}

File: frontend-expo/src/services/boardService.js
import api from './api';
import { ENDPOINTS } from '../constants/api';

export const boardService = {
    getBoards: async (userId) => {
        const response = await api.get(ENDPOINTS.GET_BOARDS, { params: { userId } });
        return response.data;
    },

    createBoard: async (boardData, userId) => {
        const response = await api.post(ENDPOINTS.CREATE_BOARD, boardData, { params: { userId } });
        return response.data;
    },

    getBoard: async (id) => {
        const response = await api.get(ENDPOINTS.GET_BOARD(id));
        return response.data;
    },

    updateBoard: async (id, boardData, userId) => {
        const response = await api.put(ENDPOINTS.UPDATE_BOARD(id), boardData, { params: { userId } });
        return response.data;
    },

    addMember: async (boardId, memberUserId, requesterUserId) => {
        const response = await api.post(
            ENDPOINTS.ADD_BOARD_MEMBER(boardId),
            { userId: memberUserId },
            { params: { userId: requesterUserId } },
        );
        return response.data;
    },
};
